samcli/lib/pipeline/bootstrap/stage.py

```diff
@@ -120,6 +142,46 @@ def _get_non_user_provided_resources_msg(self) -> str:
         )
         return "\n".join([f"\t- {comment}" for comment in resource_comments])
 
+    def _should_create_new_provider(self) -> bool:
+        if not self.oidc_provider.provider_url:
+            return False
+        iam_client = boto3.client("iam")
+        providers = iam_client.list_open_id_connect_providers()
+        url_to_compare = self.oidc_provider.provider_url.replace("https://", "")
+        for provider_resource in providers["OpenIDConnectProviderList"]:
+            if url_to_compare in provider_resource["Arn"]:
+                return False
+        return True
+
+    def _generate_thumbprint(self) -> Optional[str]:
```

Comment:
This could be static; it seems generating the thumbprint for some domain is a well-defined process that makes sense on its own. Would be helpful to also e.g. some authoritative reference that defines what a thumbprint is technically.